packages/dev/core/src/Cameras/targetCamera.ts

```diff
@@ -424,12 +424,12 @@ export class TargetCamera extends Camera {
         this._updateCameraRotationMatrix();
 
         // Apply the changed rotation to the upVector
-        if (this.rotationQuaternion && this._cachedQuaternionRotationZ != this.rotationQuaternion.z) {
+        if (this.rotationQuaternion && !this._cachedQuaternionRotation.equalsWithEpsilon(this.rotationQuaternion)) {
             this._rotateUpVectorWithCameraRotationMatrix();
-            this._cachedQuaternionRotationZ = this.rotationQuaternion.z;
-        } else if (this._cachedRotationZ !== this.rotation.z) {
+            this._cachedQuaternionRotation.copyFrom(this.rotationQuaternion);
+        } else if (this._cachedRotation.equalsWithEpsilon(this.rotation)) {

```

Comment:
missing a bang to be in the NOT case :-)

You should check if you can add tests.

Reply:
Good catch!  I checked quaternion but didn't check rotation.  I've fixed the issue and added tests.